pyspecProcScripts/calc_error.py

```diff
@@ -0,0 +1,102 @@
+from .simple_functions import select_pathway
+import numpy as np
+
+
+# {{{ Functions to calculate var and mean of nan-masked data
+#     Pulled from time_domain_noise.py example
+def _masked_mean_multi(x, axis=None):
+    "Calculate the mean of nan-masked data on a 1D axis"
+    assert axis is not None
+
+    def masked_mean(x):
+        return np.mean(x[np.isfinite(x)])
+
+    return np.apply_along_axis(masked_mean, axis, x)
+
+
+def _masked_var_multi(x, var_has_imag=False, axis=None):
+    """Calculates the variance along a 1D axis.
+    If the data is complex you must assign var_has_imag as true
+    so that the calculated variance is divided by 2.
+    By default it calculates the variance of the real of the data"""
+    assert axis is not None
+
+    def masked_var(x):
+        if var_has_imag:  # take average of variance along real and imag
+            return np.var(x[np.isfinite(x)], ddov=1) / 2
+        else:
+            return np.var(x[np.isfinite(x)], ddof=1)
+
+    return np.apply_along_axis(masked_var, axis, x)
+
+
+# }}}
+def calc_masked_error(
+    s,
+    frq_slice,
+    signal_pathway,
+    excluded_pathways=None,
+    direct="t2",
+    indirect="nScans",
+    fl=None,
+):
+    """Calculates the propagation of error for the given signal.
+
+    Before declaring the error_path,
+    look at an examples such as integration_w_error.py to see how to
+    decide which excluded pathways to take the error over.
+
+    Parameters
+    ==========
+    s: nddata
+        Full nddata set in the frequency domain for which the error is being
+        propagated.
+    frq_slice: tuple
+        Frequency range that will be filtered out in calculating the error - it
+        is assumed this region in all coherence transfer pathways contains some
+        amount of phase cycling noise.
+    signal_pathway:   dict
+                Dictionary of the path of the desired signal.
```

Comment:
I realized you  don't want to pass this separately.  Here, there is not difference between the signal pathway and any other excluded pathway

Reply:
I like having this as a default mask choice though so even if now excluded pathways are fed it will still filter out the signal pathway... still nix it?

Reply:
calc_error no longer takes signal_pathways as a kwarg